src/Symfony/Component/Form/ChoiceList/Factory/Cache/AbstractStaticOption.php

```diff
@@ -41,7 +40,7 @@ final public function __construct(FormTypeInterface|FormTypeExtensionInterface $
     {
         $hash = CachingFactoryDecorator::generateHash([static::class, $formType, $vary]);
 
-        $this->option = self::$options[$hash] ?? self::$options[$hash] = $option;
+        $this->option = self::$options[$hash] ??= \is_callable($option) && !$option instanceof \Closure ? \Closure::fromCallable($option) : $option;
```

Comment:
a faster way to write this could be
`$option instanceof \Closure || !\is_callable($option)? $option : \Closure::fromCallable($option)`